buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java

```diff
@@ -255,4 +284,102 @@ private void assertFileProcessed(String testName) {
     String declaration = "final void foo() :";
     assertFunctionNameExtracted(declaration, "foo");
   }
+
+  @Test public void testProcessKeywordsBothEmpty() {
+    int keywordsSize = 0;
+    int nonReservedKeywordsSize = 0;
+    assertKeywordsProcessed(new LinkedHashMap<Keyword, String>(),
+        new HashSet<Keyword>(), new ExtractedData(), keywordsSize,
+        nonReservedKeywordsSize);
+  }
+
+  @Test public void testProcessKeywordsBothNonEmptyAndValid() {
+    int keywordsSize = 2;
+    int nonReservedKeywordsSize = 1;
+    Map<Keyword, String> keywords = new LinkedHashMap<Keyword, String>();
+    Set<Keyword> nonReservedKeywords = new HashSet<Keyword>();
+
+    keywords.put(new Keyword("foo"), "foo");
+    keywords.put(new Keyword("bar"), "bar");
+    nonReservedKeywords.add(new Keyword("bar"));
+    assertKeywordsProcessed(keywords, nonReservedKeywords, new ExtractedData(),
+        keywordsSize, nonReservedKeywordsSize);
+  }
+
+  @Test public void testProcessKeywordsNonReservedKeywordInExtractedData() {
+    int keywordsSize = 2;
+    int nonReservedKeywordsSize = 1;
+    Map<Keyword, String> keywords = new LinkedHashMap<Keyword, String>();
+    ExtractedData extractedData = new ExtractedData();
+    Set<Keyword> nonReservedKeywords = new HashSet<Keyword>();
+
+    keywords.put(new Keyword("foo"), "foo");
+    extractedData.keywords.put(new Keyword("bar"), "bar");
+    nonReservedKeywords.add(new Keyword("bar"));
+    assertKeywordsProcessed(keywords, nonReservedKeywords, extractedData,
+        keywordsSize, nonReservedKeywordsSize);
+  }
+
+  @Test public void testProcessKeywordsKeywordGetsOverridden() {
+    int keywordsSize = 1;
+    int nonReservedKeywordsSize = 0;
+    Map<Keyword, String> keywords = new LinkedHashMap<Keyword, String>();
+    ExtractedData extractedData = new ExtractedData();
+    Set<Keyword> nonReservedKeywords = new HashSet<Keyword>();
+    extractedData.keywords.put(new Keyword("foo"), "foo1");
+    keywords.put(new Keyword("foo"), "foo2");
+    assertKeywordsProcessed(keywords, nonReservedKeywords, extractedData,
+        keywordsSize, nonReservedKeywordsSize);
+  }
+
+  @Test public void testProcessKeywordsInvalidNonReservedKeywordFails() {
+    Map<Keyword, String> keywords = new LinkedHashMap<Keyword, String>();
+    ExtractedData extractedData = new ExtractedData();
+    Set<Keyword> nonReservedKeywords = new HashSet<Keyword>();
+
+    keywords.put(new Keyword("foo"), "foo");
+    nonReservedKeywords.add(new Keyword("bar"));
+    assertKeywordsNotProcessed(keywords, nonReservedKeywords, extractedData);
```

Comment:
nit: Can we just pass "new ExtractedData()" directly in here as is being done in testProcessKeywordsBothNonEmptyAndValid()?